course_UvA-DL/inception-resnet-densenet/Inception_ResNet_DenseNet.py

```diff
@@ -0,0 +1,1298 @@
+# %% [markdown]
+# Let's start with importing our standard libraries here.
+
+# %%
+import os
+import random
+import urllib.request
+from types import SimpleNamespace
+from urllib.error import HTTPError
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+import seaborn as sns
+import tabulate
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.utils.data as data
+import torchvision
+# %matplotlib inline
+from IPython.display import display, HTML, set_matplotlib_formats
+from PIL import Image
+from pytorch_lightning.callbacks import LearningRateMonitor, ModelCheckpoint
+from torchvision import transforms
+from torchvision.datasets import CIFAR10
+
+set_matplotlib_formats('svg', 'pdf')  # For export
+matplotlib.rcParams['lines.linewidth'] = 2.0
+sns.reset_orig()
+
+# PyTorch
+# Torchvision
+
+# %% [markdown]
+# We will use the same `set_seed` function as in the previous tutorials, as well as the path variables `DATASET_PATH` and `CHECKPOINT_PATH`.
+# Adjust the paths if necessary.
+
+# %%
+# Path to the folder where the datasets are/should be downloaded (e.g. CIFAR10)
+DATASET_PATH = os.environ.get("PATH_DATASETS", "data/")
+# Path to the folder where the pretrained models are saved
+CHECKPOINT_PATH = os.environ.get("PATH_CHECKPOINT", "saved_models/ConvNets")
+
+
+# Function for setting the seed
+def set_seed(seed):
+    random.seed(seed)
+    np.random.seed(seed)
+    torch.manual_seed(seed)
+    if torch.cuda.is_available():
+        torch.cuda.manual_seed(seed)
+        torch.cuda.manual_seed_all(seed)
+
+
+set_seed(42)
+
+# Ensure that all operations are deterministic on GPU (if used) for reproducibility
+torch.backends.cudnn.determinstic = True
+torch.backends.cudnn.benchmark = False
+
+device = torch.device("cuda:0") if torch.cuda.is_available() else torch.device("cpu")
+
+# %% [markdown]
+# We also have pretrained models and Tensorboards (more on this later) for this tutorial, and download them below.
+
+# %%
+# Github URL where saved models are stored for this tutorial
+base_url = "https://raw.githubusercontent.com/phlippe/saved_models/main/tutorial5/"
```

Comment:
do you think it would make sense to have kind of releases with Tutorials and upload these models to the release?

Reply:
Wouldn't be a bad idea. The separate GitHub repo is more of a quick hack than a long-term solution.

Reply:
ok, mind open an issue for it and I ll have look at it later... just to not forget it :]